### DISCUSSION TITLE
Good practice in comparing 0/1 values from mysql in php

### DISCUSSION THREAD

**Discussion by ghost** [laravel/framework#53916] [General]
Take this type of column for example

`is_enabled enum('1','0') COLLATE utf8mb3_unicode_ci DEFAULT '0',`

It will play tricks on you if you update it to 0 as int in mysql 8 ( It will become empty string).

The laravel migration with boolean converts it like:

`is_enabled tinyint(1) DEFAULT '0',`

In both cases this is what saved me from having to fix bugs in code:

```
if ((string)$model->is_enabled !== '1') {
}

if ((string)$model->is_enabled === '1') {
}
```

So never compare with anything but 1 as string.

UPDATE
the db migration converted '' to 0, '1' to 1 and '0' to, pay attention 2 :))))

**Comment by ahinkle** [comment]
Have you tried using an [Attribute Cast](https://laravel.com/docs/11.x/eloquent-mutators#attribute-casting) of `boolean`?

**Reply by ghost** [reply]
The db was enum string. False would become empty string anyway.

**Reply by ghost** [reply]
I have a golden rule. From mysql I can receive, **int, string, null**. All my comparisons are converted to strings just like in the above example to avoid type differences (or handled as null).

**Reply by ahinkle** [reply]
That's definitely not a typical approach, but hey, to each their own. 🙂

**Reply by ghost** [reply]
I see casts as unnecessary for mysql at least. Casting to string will not change the value except for null. Cast to int can change it (strings become 0).
Also avoiding casts makes the list faster especially when downloading hundred thousands rows.

**Reply by henzeb** [reply]
if you do it in query: be sure to do an actual comparison. `where is_enabled` is not using indexes, but `where is_enabled=1` does.

**Reply by ghost** [reply]
@henzeb Thanks. I did not knew that. But I never use where like that except for ` where 1 `or `where 0.`
